internal/datastore/crdb/crdb.go

```diff
@@ -177,6 +182,7 @@ func newCRDBDatastore(url string, options ...Option) (datastore.Datastore, error
 		writePool,
 		config.watchBufferLength,
 		keyer,
+		keySetInit,
```

Comment:
instead of passing this in as a HOF, could we instead have a function that always gets called that takes the context and the keyer and then does a `switch` over the keyer?

Reply:
I wanted to have the `keyer` no-op immediately if you're providing the key in the request, since it gets called for every namespace touched on write.